architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md

```diff
@@ -0,0 +1,230 @@
+## Authoring a User-Defined-Type schema
+
+* **Author**: Reshma Abdul Rahim (@reshrahim)
+
+## Summary
+<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->
+User-Defined Types enable Platform engineers to define and deploy their organization's custom services in Radius. More details around the feature, user scenarios and end-end-experience can be found [here](/architecture/2024-06-resource-extensibility-feature-spec.md). This feature specification document details on the requirements and the user experience for authoring and updating the schema for the User-Defined-Type.
+
+### Top level goals
+<!-- At the most basic level, what are we trying to accomplish? -->
+
+- Enable a simple and intuitive experience for platform engineers to define the schema for their User-Defined-Type with minimal lines of code
```

Comment:
Concept count is better than lines of code. There's all kinds of wierd games to play if LOC is the thing we're optimizing.